moda-contracts/test/SplitsFactoryWithFork.t.sol

```diff
@@ -0,0 +1,198 @@
+    address _splitMainAddress = 0x2ed6c4B5dA6378c7897AC67Ba9e43102Feb694EE;
+    address _payer = makeAddr("MODA_payer");
+    address _modaTreasury = makeAddr("MODA_treasury");
+    address _deployer = makeAddr("MODA_deployer");
+    address _alice = makeAddr("MODA_user_alice");
+    address _bob = makeAddr("MODA_user_bob");
+    address _distributor = makeAddr("MODA_distributor");
+
+    address _split;
+
+    uint32 _modaFee = 1_000;
+    uint32 _modaFeeMultiplier = 10_000;
+
+    address[] _beneficiaries;
+
+    event SplitCreated(address indexed sender, address indexed split);
+
+    function _sortAddresses(address[] memory addresses) internal pure returns (address[] memory) {
+        uint256 length = addresses.length;
+
+        for (uint256 i = 0; i < length; i++) {
+            for (uint256 j = 0; j < length - i - 1; j++) {
+                if (addresses[j] > addresses[j + 1]) {
+                    // Swap
+                    address temp = addresses[j];
+                    addresses[j] = addresses[j + 1];
+                    addresses[j + 1] = temp;
+                }
+            }
+        }
+
+        return addresses;
+    }
+
+    function setUp() public {
+        uint256 evmFork = vm.createFork(EVM_RPC_URL);
+        vm.selectFork(evmFork);
+
+        vm.deal(_payer, 10 ether);
+
+        vm.startPrank(_deployer);
+        _modaRegistry = new ModaRegistry(_modaTreasury, _modaFee, _modaFeeMultiplier);
+        _splitMain = ISplitMain(_splitMainAddress);
+        _splitsFactory = new SplitsFactory(_splitMain, _modaRegistry);
+        vm.stopPrank();
+    }
+
+    function setupBeneficiaries() public {
+        _beneficiaries.push(_alice);
+        _beneficiaries.push(_bob);
+    }
+
+    // create
+
+    function create_split_setUp() public {
+        setupBeneficiaries();
+        vm.startPrank(_deployer);
+        _split = _splitsFactory.create(_beneficiaries);
+        vm.stopPrank();
+    }
+
+    function pay_split_setUp() public {
+        create_split_setUp();
+        vm.startPrank(_payer);
+        payable(_split).transfer(10 ether);
+        vm.stopPrank();
+    }
+
+    function test_distributeETH_from_SplitMain() public {
+        pay_split_setUp();
+
+        _beneficiaries.push(_modaTreasury);
+
+        _beneficiaries = _sortAddresses(_beneficiaries);
+
+        uint32[] memory percentAllocations = new uint32[](3);
+
+        for (uint256 i = 0; i < _beneficiaries.length; i++) {
+            if (_beneficiaries[i] == _modaTreasury) {
+                percentAllocations[i] = 100_000;
+            } else {
+                percentAllocations[i] = 450_000;
+            }
+        }
+
+        uint256 gasBefore = gasleft();
+
+        vm.startPrank(_distributor);
+        _splitMain.distributeETH(_split, _beneficiaries, percentAllocations, 10_000, _distributor);
+        vm.stopPrank();
+
+        uint256 gasAfter = gasleft();
+
+        console2.log("Gas used: %s", gasBefore - gasAfter);
```

Comment:
can we add an assertion here, or is `gas` a variable based on the forked state of Polygon?